### ISSUE TITLE
DataSelectField in a block not working when instantiated for the first time editing a Page in the Manager

### ISSUE THREAD

**Issue by horacioj** [PiranhaCMS/piranha.core#1532]
Using Piranha 8.4.2

Let's say there is a block with a field defined like this:

```
public DataSelectField<LocationItem> Location { get; set; }
```

`LocationItem` implements `GetById()` and `GetList()` having a service as a parameter. example:

```
public static async Task<LocationItem> GetById(string id, ILocationData data)
```

If you have this field in a block, the first time you add this block to a page in the Manager, the `DataSelectField` isn't rendered (the dropdown) and it throws this error in the JS console:

```
piranha-deps-dev.js:16559 [Vue warn]: Error in render: "TypeError: Cannot read property '$values' of null"

found in

---> <DataSelectField>
       <GenericBlock>
         <Root>
warn @ piranha-deps-dev.js:16559
logError @ piranha-deps-dev.js:17818
globalHandleError @ piranha-deps-dev.js:17813
handleError @ piranha-deps-dev.js:17773
Vue._render @ piranha-deps-dev.js:19472
updateComponent @ piranha-deps-dev.js:19986
get @ piranha-deps-dev.js:20397
Watcher @ piranha-deps-dev.js:20386
mountComponent @ piranha-deps-dev.js:19993
Vue.$mount @ piranha-deps-dev.js:24963
Vue.$mount @ piranha-deps-dev.js:27848
init @ piranha-deps-dev.js:19050
createComponent @ piranha-deps-dev.js:21892
createElm @ piranha-deps-dev.js:21839
createChildren @ piranha-deps-dev.js:21967
createElm @ piranha-deps-dev.js:21868
createChildren @ piranha-deps-dev.js:21967
createElm @ piranha-deps-dev.js:21868
createChildren @ piranha-deps-dev.js:21967
createElm @ piranha-deps-dev.js:21868
patch @ piranha-deps-dev.js:22389
Vue._update @ piranha-deps-dev.js:19865
updateComponent @ piranha-deps-dev.js:19986
get @ piranha-deps-dev.js:20397
Watcher @ piranha-deps-dev.js:20386
mountComponent @ piranha-deps-dev.js:19993
Vue.$mount @ piranha-deps-dev.js:24963
Vue.$mount @ piranha-deps-dev.js:27848
init @ piranha-deps-dev.js:19050
createComponent @ piranha-deps-dev.js:21892
createElm @ piranha-deps-dev.js:21839
createChildren @ piranha-deps-dev.js:21967
createElm @ piranha-deps-dev.js:21868
createChildren @ piranha-deps-dev.js:21967
createElm @ piranha-deps-dev.js:21868
addVnodes @ piranha-deps-dev.js:22020
updateChildren @ piranha-deps-dev.js:22147
patchVnode @ piranha-deps-dev.js:22233
updateChildren @ piranha-deps-dev.js:22107
patchVnode @ piranha-deps-dev.js:22233
updateChildren @ piranha-deps-dev.js:22107
patchVnode @ piranha-deps-dev.js:22233
updateChildren @ piranha-deps-dev.js:22107
patchVnode @ piranha-deps-dev.js:22233
updateChildren @ piranha-deps-dev.js:22107
patchVnode @ piranha-deps-dev.js:22233
patch @ piranha-deps-dev.js:22394
Vue._update @ piranha-deps-dev.js:19868
updateComponent @ piranha-deps-dev.js:19986
get @ piranha-deps-dev.js:20397
run @ piranha-deps-dev.js:20472
flushSchedulerQueue @ piranha-deps-dev.js:20230
(anonymous) @ piranha-deps-dev.js:17914
flushCallbacks @ piranha-deps-dev.js:17840
Promise.then (async)
timerFunc @ piranha-deps-dev.js:17867
nextTick @ piranha-deps-dev.js:17924
queueWatcher @ piranha-deps-dev.js:20322
update @ piranha-deps-dev.js:20462
notify @ piranha-deps-dev.js:16670
mutator @ piranha-deps-dev.js:16822
(anonymous) @ piranha.pageedit.min.js:1
Promise.then (async)
addBlock @ piranha.pageedit.min.js:1
select @ piranha.min.js:1
click @ VM4521:3
invokeWithErrorHandling @ piranha-deps-dev.js:17788
invoker @ piranha-deps-dev.js:18113
original._wrapper @ piranha-deps-dev.js:23466
Show 24 more frames
piranha-deps-dev.js:17822 TypeError: Cannot read property '$values' of null
    at Proxy.eval (eval at createFunction (piranha-deps-dev.js:27553), <anonymous>:3:483)
    at VueComponent.Vue._render (piranha-deps-dev.js:19470)
    at VueComponent.updateComponent (piranha-deps-dev.js:19986)
    at Watcher.get (piranha-deps-dev.js:20397)
    at new Watcher (piranha-deps-dev.js:20386)
    at mountComponent (piranha-deps-dev.js:19993)
    at VueComponent.Vue.$mount (piranha-deps-dev.js:24963)
    at VueComponent.Vue.$mount (piranha-deps-dev.js:27848)
    at init (piranha-deps-dev.js:19050)
    at createComponent (piranha-deps-dev.js:21892)
```

If you save the page, the next time it works fine. I.e. the dropdown is rendered, throws not errors, and you can select a value.

The problem doesn't happen with the field in a Region. It happens on blocks only.

I see in  \Piranha.Manager\assets\src\js\components\fields\data-select-field.vue:

```
<option v-for="(item) in model.items.$values" v-bind:key="item.id" v-bind:value="item.id">
```

It looks like `model.items` isn't being populated the first time the block is added to the page.
Furthermore, `GetList()` isn't being called.
However,  `GetList()` is being called when editing the page with the block already there. Hence, no error and works fine.

In summary, the problem seems to be `GetList()` isn't being called when you editing a page in the Manager add a block having a `DataSelectField`.

**Comment by tidyui** [comment]
Duplicate of #1420. Fix for this is available in the `rc` for version `9.0`
